utests/test_cudafft.c

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2019. Christian Holme.
+ * All rights reserved. Use of this source code is governed by
+ * a BSD-style license which can be found in the LICENSE file.
+ *
+ * Authors:
+ * 2019 Christian Holme <christian.holme@med.uni-goettingen.de>
+ */
+
+
+
+
+#include <complex.h>
+#include <assert.h>
+
+#include "num/fft.h"
+#include "num/rand.h"
+
+#include "num/multind.h"
+#include "num/flpmath.h"
+
+#include "misc/debug.h"
+#include "misc/misc.h"
+
+#include "utest.h"
+
+enum { N = 6 };
+
+
+
+static bool run_cuda_fft_test (const long* dims, const unsigned int flags,
+			       const complex float* in, complex float* cpu_inout,
+			       complex float* gpu_inout, complex float* gpu_result)
+{
+	md_copy(N, dims, cpu_inout, in, CFL_SIZE);
+	md_copy(N, dims, gpu_inout, in, CFL_SIZE);
+
+	// create fft plan
+	const struct operator_s* fftplan = fft_create(N, dims, flags, cpu_inout, cpu_inout, false);
+
+	fft_exec(fftplan, cpu_inout, cpu_inout);
+	fft_exec(fftplan, gpu_inout, gpu_inout);
+
+	md_copy(N, dims, gpu_result, gpu_inout, CFL_SIZE);
+
+	UT_ASSERT(md_znrmse(N, dims, cpu_inout, gpu_result) < UT_TOL);
+
+	fft_free(fftplan);
+}
+
+
+
+
+static bool test_cuda_fft (void)
+{
+#ifdef USE_CUDA
```

Comment:
as most code as possible should be compiled always and not hidden behind ifdefs to get more comprehensive checking